### PR TITLE
kube-controller-manager: readjust log verbosity

### LINKED ISSUE #115574: cronjob controller: reconsider log verbosity
### What happened?

We are running Kubernetes 1.25.5 and trying to configure kube-controller-manager to be less verbose when logging.
We run kube-controller-manager as a static pod created by Kubelet and use the following command where we pass `--v=0`:
```
spec:
  containers:
  - command:
    - kube-controller-manager
    - --feature-gates=EphemeralContainers=true,TopologyAwareHints=true
    - --allocate-node-cidrs=true
    - --authentication-kubeconfig=/etc/kubernetes/config/controller-manager.conf
    - --authorization-kubeconfig=/etc/kubernetes/config/controller-manager.conf
    - --cluster-cidr=10.6.0.0/16
    - --configure-cloud-routes=false
    - --kubeconfig=/etc/kubernetes/config/controller-manager.conf
    - --leader-elect=true
    - --node-monitor-grace-period=120s
    - --root-ca-file=/etc/kubernetes/ssl/ca.pem
    - --service-account-private-key-file=/etc/kubernetes/ssl/signing-key.pem
    - --use-service-account-credentials
    - --v=0
    image: registry.k8s.io/kube-controller-manager:v1.25.5
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 3
      httpGet:
        host: 127.0.0.1
        path: /healthz
        port: 10257
        scheme: HTTPS
      initialDelaySeconds: 15
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 15
    name: kube-controller-manager
```
The `--v=0` flag seems to passed fine to the final command line of kube-controller-manager:
```
root     2464651  285  1.6 1939088 1085624 ?     Ssl  05:59 544:37 kube-controller-manager --feature-gates=EphemeralContainers=true,TopologyAwareHints=true --allocate-node-cidrs=true --authentication-kubeconfig=/etc/kubernetes/config/controller-manager.conf --authorization-kubeconfig=/etc/kubernetes/config/controller-manager.conf --cluster-cidr=10.6.0.0/16 --configure-cloud-routes=false --kubeconfig=/etc/kubernetes/config/controller-manager.conf --leader-elect=true --node-monitor-grace-period=120s --root-ca-file=/etc/kubernetes/ssl/ca.pem --service-account-private-key-file=/etc/kubernetes/ssl/signing-key.pem --use-service-account-credentials --v=0
```

Even though we get tons of the following logs periodically:
```
kube-controller-manager-master-2.dev.merit.uw.systems kube-controller-manager I0202 06:52:05.808355       1 event.go:294] "Event occurred" object="energy/gba-fabricator-job" fieldPath="" kind="CronJob" apiVersion="batch/v1" type="Normal" reason="JobAlreadyActive" message="Not starting job because prior execution is running and concurrency policy is Forbid"
```
which come from https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cronjob/cronjob_controllerv2.go#L575 where is logged as klog.V(4).InfoS
So arguably this line should not appear on our controller logs.

### What did you expect to happen?

Setting flag `--v=0` should be enough to lower log verbosity level according to: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md 

### How can we reproduce it (as minimally and precisely as possible)?

Scheduling a long running job at small intervals should be enough to reproduce our exact case and get the above mentioned logs.

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version --short
Client Version: v1.25.5
Kustomize Version: v4.5.7
Server Version: v1.25.5
```

</details>

### Cloud provider

<details>
On premises
</details>

### OS version

<details>

```console
$ cat /etc/os-release
NAME="Flatcar Container Linux by Kinvolk"
ID=flatcar
ID_LIKE=coreos
VERSION=3374.2.3
VERSION_ID=3374.2.3
BUILD_ID=2023-01-20-1751
SYSEXT_LEVEL=1.0
PRETTY_NAME="Flatcar Container Linux by Kinvolk 3374.2.3 (Oklo)"
ANSI_COLOR="38;5;75"
HOME_URL="https://flatcar.org/"
BUG_REPORT_URL="https://issues.flatcar.org"
FLATCAR_BOARD="amd64-usr"
CPE_NAME="cpe:2.3:o:flatcar-linux:flatcar_linux:3374.2.3:*:*:*:*:*:*:*"

$ uname -a
Linux worker-54.dev.merit.uw.systems 5.15.86-flatcar #1 SMP Fri Jan 20 17:26:55 -00 2023 x86_64 Intel(R) Xeon(R) CPU E3-1585L v5 @ 3.00GHz GenuineIntel GNU/Linux
```

</details>

### Install tools

<details>
Self managed Kubernetes cluster.
</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by mengjiao-liu**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Increase the global level for broadcaster's logging to 3 so that users can ignore event messages by lowering the logging level. It reduces information noise.

refer to https://github.com/kubernetes/kubernetes/issues/115574#issuecomment-1806453969

> Looking at this issue more in depth there are 2 things that I'd like to see being resolved:
>
> Increasing the global level for broadcaster's logging to 3, such that this is the new base line. See [here](https://github.com/kubernetes/kubernetes/blob/1f3256bbac668f120433990028ca1fe206e4117b/staging/src/k8s.io/client-go/tools/record/event.go#L368-L372) for more details.
Making sure the context is properly injected into the broadcaster, this will allow the -v flag value to be used also in that broadcaster, rather than the above global value. Again see [here](https://github.com/kubernetes/kubernetes/blob/1f3256bbac668f120433990028ca1fe206e4117b/staging/src/k8s.io/client-go/tools/record/event.go#L368-L372) for explanation.
In other words, looking at cronjob controller we need:
>
> [this line](https://github.com/kubernetes/kubernetes/blob/1f3256bbac668f120433990028ca1fe206e4117b/pkg/controller/cronjob/cronjob_controllerv2.go#L132) needs to look like:
jm.broadcaster.StartStructuredLogging(3)
[this line](https://github.com/kubernetes/kubernetes/blob/1f3256bbac668f120433990028ca1fe206e4117b/pkg/controller/cronjob/cronjob_controllerv2.go#L85) need to look like:
eventBroadcaster := record.NewBroadcaster(record.WithContext(ctx))

related:  https://github.com/kubernetes/enhancements/issues/3077
> Besides migrating log calls, we also might have to migrate from APIs which don't support contextual logging to APIs which do:
>
 >k8s.io/client-go/tools/[events|record]: https://github.com/kubernetes/kubernetes/pull/120729
>https://github.com/kubernetes/kubernetes/issues/115574

---
Take the `cronjob` controller integration test(
test/integration/cronjob/cronjob_test.go) output results as an example:
- set -v=0 or 2 (<3):
```
go test -v ./test/integration/cronjob/ -run TestCronJobLaunchesPodAndCleansUp -args -v=0 |grep  "Event occurred"

go test -v ./test/integration/cronjob/ -run TestCronJobLaunchesPodAndCleansUp -args -v=2 |grep  "Event occurred"

```
We can see that there is no event output, which is what we expect.

- set -v=3 (>=3):
```
go test -v ./test/integration/cronjob/ -run TestCronJobLaunchesPodAndCleansUp -args -v=3|grep  "Event occurred"
    event.go:376: I1220 17:18:00.123012] Event occurred object="simple-cronjob-test/foo" fieldPath="" kind="CronJob" apiVersion="batch/v1" type="Normal" reason="SuccessfulCreate" message="Created job foo-28384398"
    event.go:376: I1220 17:18:00.263791] Event occurred object="simple-cronjob-test/foo-28384398" fieldPath="" kind="Job" apiVersion="batch/v1" type="Normal" reason="SuccessfulCreate" message="Created pod: foo-28384398-h5x7k"
```
We can see that the event now has output results.

#### Which issue(s) this PR fixes:

Fixes #115574

#### Special notes for your reviewer:
Now, the NewBroadcaster function with ctx creates a new goroutine to call `eventBroadcaster.Broadcaster.Shutdown()`, so creating a new controller needs to cancel ctx to avoid goroutine leakage in the integration tests.
https://github.com/kubernetes/kubernetes/blob/7897910469aa091ebf6576740d055a7137fa147c/staging/src/k8s.io/client-go/tools/record/event.go#L205-L207
see details for https://github.com/kubernetes/kubernetes/pull/120729.

#### Does this PR introduce a user-facing change?

```release-note
kube-controller-manager: increase the global level for broadcaster's logging to 3 so that users can ignore event messages by lowering the logging level. It reduces information noise.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/sig apps
/sig instrumentation

### CODE REVIEW

**Comment by mengjiao-liu** [comment]
pull-kubernetes-e2e-autoscaling-hpa-cm failed due to an issue reported by https://github.com/kubernetes/kubernetes/issues/121738, not related to this PR.

**Comment by mengjiao-liu** [comment]
ping @soltysh

File: pkg/controller/endpoint/endpoints_controller_test.go [hunk 1]
```diff
@@ -1671,15 +1786,17 @@ func TestPodUpdatesBatching(t *testing.T) {
 			resourceVersion := 1
 			testServer, endpointsHandler := makeTestServer(t, ns)
 			defer testServer.Close()
-			endpoints := newController(testServer.URL, tc.batchPeriod)
-			stopCh := make(chan struct{})
-			defer close(stopCh)
+
+			_, ctx := ktesting.NewTestContext(t)
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
```
Comment (pohly) [inline pkg/controller/endpoint/endpoints_controller_test.go:1792]: The PR looks good, but can we do it after https://github.com/kubernetes/kubernetes/pull/122419/ plus the additional commit mentioned in https://github.com/kubernetes/kubernetes/pull/122419/commits/9ef90353036bcc6efd7c41a24712ba6352f2ab54#r1478192511 are merged?

This this code here becomes simply:
```
tCtx := ktesting.Init(t)
```

Cancellation is going to be handled by the revised `ktesting`.

Comment (mengjiao-liu) [reply pkg/controller/endpoint/endpoints_controller_test.go:1792]: Ok, I can wait for PR #122419 to merge and update the PR to use the new function.

**Comment by mengjiao-liu** [comment]
/hold 
Wait for PR https://github.com/kubernetes/kubernetes/pull/122419 to merge.

**Comment by mengjiao-liu** [comment]
> /hold Wait for PR #122419 to merge.

PR #122419 was merged.
/hold cancel 
Changed as suggested in the comment https://github.com/kubernetes/kubernetes/pull/122293#discussion_r1478295105.

**Comment by mengjiao-liu** [comment]
Related issue: https://github.com/kubernetes/kubernetes/issues/122141

**Comment by mengjiao-liu** [comment]
https://github.com/kubernetes/kubernetes/pull/122293/commits/0d54825060bdd2127261ad5f52bf2802315f33fe modified content has been updated by has been updated by pr https://github.com/kubernetes/kubernetes/pull/123444/commits/840ef149071c8c433fc9c03ae14d6efc387a1c83, remove this commit of this PR.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**